### PR TITLE
chore: Change Query/DenomTrace gRPC

### PR DESCRIPTION
**Pull request by catShaark**
## Description

Change the `DenomTrace` grpc so that it takes the whole ibc denom instead of just the hash

closes: #101

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/transfer/keeper/grpc_query.go [hunk 1]
```diff
@@ -22,17 +23,21 @@ func (q Keeper) DenomTrace(c context.Context, req *types.QueryDenomTraceRequest)
...
 
-	hash, err := types.ParseHexHash(req.Hash)
+	if !strings.HasPrefix(req.Denom, "ibc/") {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid denom with no ibc prefix %s", req.Denom))
+	}
+
+	hash, err := types.ParseHexHash(req.Denom[4:])
 	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid denom trace hash %s, %s", req.Hash, err))
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid denom trace hash %s, %s", req.Denom[4:], err))
```
Comment (crodriguezvega) [inline modules/apps/transfer/keeper/grpc_query.go:32]:
```suggestion
		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid denom trace hash: %s, error: %s", req.Denom[4:], err))
```

**Review by crodriguezvega — COMMENTED** [review]
Thanks, @catShaark. I left just a couple of nits. And we need a changelog entry. :) Since this changing the API, then I would say in the API breaking section.

File: modules/apps/transfer/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -21,10 +21,10 @@ func (suite *KeeperTestSuite) TestQueryDenomTrace() {
 		expPass  bool
 	}{
 		{
-			"invalid hex hash",
+			"invalid ibc denom",
```
Comment (seantking) [inline modules/apps/transfer/keeper/grpc_query_test.go:24]: nit: could we put the success case first? :=)

Comment (catShaark) [reply modules/apps/transfer/keeper/grpc_query_test.go:24]: yesss

**Review by seantking — APPROVED** [review]
Great work!

File: modules/apps/transfer/keeper/grpc_query.go [hunk 3]
```diff
@@ -22,17 +23,21 @@ func (q Keeper) DenomTrace(c context.Context, req *types.QueryDenomTraceRequest)
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
-	hash, err := types.ParseHexHash(req.Hash)
+	if !strings.HasPrefix(req.Denom, "ibc/") {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid denom with no ibc prefix: %s", req.Denom))
+	}
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/grpc_query.go:28]: I think we should make this change backwards compatible. We could do so by checking if it has the ibc prefix and then remove the prefix before performing the query. I should have specified this on the original issue (but at the time there was a lot less dependencies on this gRPC

So essentially:
```go
if strings.HasPrefix(denom, "ibc/") {
    denom = denom[4:]
}

Comment (catShaark) [reply modules/apps/transfer/keeper/grpc_query.go:28]: So I'll make it like you said and leave the field name still be `denom`

Comment (crodriguezvega) [reply modules/apps/transfer/keeper/grpc_query.go:28]: That's a good suggestion from @colin-axner. If you make the change backwards compatible, then the changelog entry shouldn't be in the API breaking section, but in the improvements one instead.

Comment (catShaark) [reply modules/apps/transfer/keeper/grpc_query.go:28]: I'll apply the change accordingly

**Comment by catShaark** [comment]
I'll close this PR and make a new one

**Pull request closed without merging**